docs/source/configuration/persisted-queries.mdx

```diff
@@ -1,72 +1,164 @@
+
+1. Preregister trusted operations from your clients
+2. Configure your router to enable safelisting
+
+This article details the latter.
+
+For more information on other configuration aspects, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries).
+
+## Router configuration
+
+After client teams publish operations to a graph's PQL, you can configure router instances to fetch and use the PQL by following these steps:
+
+1. Ensure your router instances are ready to work with PQLs:
+    - Make sure you're using version `1.25.0` or later of the Apollo Router.
+    - Make sure your router instances are [connected to your GraphOS Enterprise organization](/router/enterprise-features/#enabling-enterprise-features) and that they're associated with a variant that your PQL is linked to.
+
+2. Set your desired security level in your router's YAML config file. For supported options, see [router security levels](#router-security-levels).
+
+3. Deploy your updated router instances to begin using your PQL.
+
+### Router security levels
+
+As you may recall, the Apollo Router supports the following [security levels](#security-levels), in increasing order of restrictiveness:
+
+- **Performance-only**: Clients can optionally execute an operation on your router by providing the operation's PQL-specified ID.
+    - All other levels also provide this core capability.
+    - This level provides no operation safelisting.
+- **Audit**: Executing operations by providing a PQL-specified ID is still optional, but the router also logs any unregistered operations.
+    - The level serves as a dry run and helps you identify operations you may still need to register before turning on safelisting.
+- **Safelisting with operation strings allowed**: The router _rejects_ any incoming operations that aren't present in its PQL.
+    - Before moving to this security level, ensure _all_ your client operations are present in your PQL.
+- **Safelisting with IDs only**: The router rejects any incoming operations that aren't present in its PQL, _and_ clients _must_ execute an operation by providing its PQL-specified ID.
+    - Before moving to this security level, ensure _all_ your clients execute operations by providing their PQL-specified ID.
+
+When you first adopt persisted queries, you should start with a less restrictive security level—either [performance-only](#performance-only) or [audit](#audit-dry-run). You can then enable increasingly restrictive levels after you update all your clients.
+
+See below for sample YAML configurations for each level.
+
+#### Performance-only
+
+To use persisted queries _only_ to reduce network bandwidth and latency (_not_ for safelisting), add the following minimal configuration:
 
 ```yaml title="router.yaml"
 preview_persisted_queries:
   enabled: true
-  log_unknown: true
-  safelist:
-    require_id: true
-    enabled: true
+apq:
+  enabled: false # APQ must be disabled
```

Comment:
Requiring APQ be off was in the docs draft I inherited. However, I noticed this wasn't included in the config samples in https://github.com/apollographql/router/pull/3484/files.

@BrynCooke , can you confirm if users can have both apq and pq simultaneously enabled in the router?

Reply:
Clarified with Adam that you must turn off APQ only for the two safelisiting modes.